code/modules/mob/living/emote.dm

```diff
@@ -1,344 +1,429 @@
+
+/* EMOTE DATUMS */
+/datum/emote/living
+	mob_type_allowed_typecache = list(/mob/living)
+	mob_type_blacklist_typecache = list(/mob/living/simple_animal/slime, /mob/living/brain)
+
+/datum/emote/living/aflap
+	key = "aflap"
+	message = "flaps their wings ANGRILY!"
+	restraint_check = TRUE
+
+/datum/emote/living/blush
+	key = "blush"
+	key_third_person = "blushes"
+	message = "blushes."
+
+/datum/emote/living/bow
+	key = "bow"
+	key_third_person = "bows"
+	message = "bows."
+	message_param = "bows to %t."
+
+/datum/emote/living/burp
+	key = "burp"
+	key_third_person = "burps"
+	message = "burps."
+	emote_type = EMOTE_AUDIBLE
+
+/datum/emote/living/choke
+	key = "choke"
+	key_third_person = "chokes"
+	message = "chokes!"
+	emote_type = EMOTE_AUDIBLE
+
+/datum/emote/living/cross
+	key = "cross"
+	key_third_person = "crosses"
+	message = "crosses their arms."
+
+/datum/emote/living/chuckle
+	key = "chuckle"
+	key_third_person = "chuckles"
+	message = "chuckles."
+	emote_type = EMOTE_AUDIBLE
+
+/datum/emote/living/collapse
+	key = "collapse"
+	key_third_person = "collapses"
+	message = "collapses!"
+	emote_type = EMOTE_AUDIBLE
+
+/datum/emote/living/collapse/run_emote(mob/user, params)
+	. = ..()
+	user.Paralyse(2)
+
+/datum/emote/living/cough
+	key = "cough"
+	key_third_person = "coughs"
+	message = "coughs!"
+	emote_type = EMOTE_AUDIBLE
+
+/datum/emote/living/dance
+	key = "dance"
+	key_third_person = "dances"
+	message = "dances around happily."
+	restraint_check = TRUE
+
+/datum/emote/living/deathgasp
+	key = "deathgasp"
+	key_third_person = "deathgasps"
+	message = "seizes up and falls limp, their eyes dead and lifeless..."
+	message_robot = "memes up and die."
+	message_alien = "lets out a waning guttural screech, green blood bubbling from its maw..."
+	message_larva = "lets out a sickly hiss of air and falls limply to the floor..."
+	message_monkey = "lets out a faint chimper as it collapses and stops moving..."
+
+/datum/emote/living/deathgasp/run_emote(mob/user, params)
+	. = ..()
+	if(. && isalienadult(user))
+		playsound(user.loc, 'sound/voice/hiss6.ogg', 80, 1, 1)
+
+/datum/emote/living/drool
+	key = "drool"
+	key_third_person = "drools"
+	message = "drools."
+
+/datum/emote/living/faint
+	key = "faint"
+	key_third_person = "faints"
+	message = "faints."
+
+/datum/emote/living/faint/run_emote(mob/user, params)
+	. = ..()
+	if(!user.sleeping)
+		user.SetSleeping(10)
+
+/datum/emote/living/flap
+	key = "flap"
+	key_third_person = "flaps"
+	message = "flaps their wings."
+	restraint_check = TRUE
+
+/datum/emote/living/flap/run_emote(mob/user, params)
+	. = ..()
+	if(ishuman(user))
+		var/mob/living/carbon/human/H = user
+		if((H.dna.features["wings"] != "None") && !("wings" in H.dna.species.mutant_bodyparts))
+			H.OpenWings()
+			addtimer(H, "CloseWings", 20)
+
+/datum/emote/living/flip
+	key = "flip"
+	key_third_person = "flips"
+
+/datum/emote/living/flip/run_emote(mob/user, params)
+	. = ..()
+	if(!user.restrained() || !user.sleeping)

```

Comment:
Honestly I'm just killing the restraint check for *flip. Not really an issue if a mob is flipping while restrained.